### PR TITLE
sys/base64: Implement Base 64 Encoding with URL and Filename Safe Alphabet

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

RFC4648 specifies an alternate alphabet for base64 encoding / decoding where '+' and '/' are exchanged for '-' and '_' to make the resulting string safe to use in filenames and URLs.

This adds a `base64url_encode()` function that uses the alternate alphabet.

The `base64_decode()` function is extended to accept both alphabets.

### Testing procedure

Run 

    make -C tests/unittests tests-base64

### Issues/PRs references

https://tools.ietf.org/html/rfc4648#page-7

### CODE REVIEW

**Comment by benpicco** [comment]
> Would it make this rather compile-time based with a `CONFIG_BASE64_URLSAFE` macro?

I thought about adding a `base64url` pseudo-module.
On `same54-xpro` this PR adds 64 bytes of ROM to `tests-base64` (without the new test).

**Comment by miri64** [comment]
> I thought about adding a `base64url` pseudo-module.
> On `same54-xpro` this PR adds 64 bytes of ROM to `tests-base64` (without the new test).

Compared to what? Shouldn't it, if implemented correctly have exactly the same size as before?

**Comment by benpicco** [comment]
Compared to master - I did build the test with and without c27de35c9bfece4f34daa68181af9fdeadde9792.

Turns out the compiler didn't optimize away the `urlsafe` parameter in `getsymbol`.
When always setting it to `false` if the `base64url` module is not used, the code size remains the same.

**Comment by miri64** [comment]
> Turns out the compiler didn't optimize away the `urlsafe` parameter in `getsymbol`.

But when you are making it compile-time configurable you would not need that, right?

**Comment by benpicco** [comment]
Now it *is* compile-time configurable.

**Comment by miri64** [comment]
But now you only can optimize-out the non-URL-safe version, right?

**Comment by benpicco** [comment]
Yes, but I think adding more complexity to this is in no relation to the code size we might save here.

File: sys/include/base64.h [hunk 1]
```diff
@@ -76,6 +76,30 @@ static inline size_t base64_estimate_encode_size(size_t data_in_size)
 int base64_encode(const void *data_in, size_t data_in_size,
                   unsigned char *base64_out, size_t *base64_out_size);
 
+/**
+ * @brief           Encodes a given datum to base64 with URL and Filename Safe Alphabet
+ *                  and save the result to the given destination.
+ *
+ * @note            Requires the use of the `base64url` module.
+ *
```
Comment (miri64) [inline sys/include/base64.h]:
```suggestion
 *
 * @see [RFC 4648, section 5](https://tools.ietf.org/html/rfc4648#section-5)
 *
```

Comment (benpicco) [reply sys/include/base64.h]: added.

**Pull request merged**